internal/cloud/backend_plan.go

```diff
@@ -235,6 +238,10 @@ in order to capture the filesystem context the remote workspace expects:
 	if configDiags.HasErrors() {
 		return nil, fmt.Errorf("error loading config with snapshot: %w", configDiags.Errs()[0])
 	}
+	if err := b.AssertImportCompatible(config); err != nil {
```

Comment:
Unfortunately this check will not run on the agent! The `cloud` backend delegates to the `local` backend to perform actual operations and the handoff happens [much earlier](https://github.com/hashicorp/terraform/blob/ea0ebcdc05beecc2e8f5e18578af5374ca2b1705/internal/cloud/backend.go#L686-L691) - in its current location this check will only run when a user is running the CLI locally, when it will never fail 🙃 

I think we'll likely need to move this check into the `command` package somewhere to have it actually run when the `cloud` backend is running on the agent. I noodled a bit with the idea of having it run during `init`, since TFC operations always init prior to the plan - perhaps something along these lines? https://github.com/hashicorp/terraform/pull/33290/files

Reply:
Yes, that is a much better idea given the wrong assumptions in this PR. Closing.